ArduCopter/compassmot.cpp

```diff
@@ -42,7 +42,7 @@ MAV_RESULT Copter::mavlink_compassmot(const GCS_MAVLINK &gcs_chan)
 
     // check compass health
     compass.read();
-    for (uint8_t i=0; i<compass.get_count(); i++) {
+    for (uint8_t i=0; i<COMPASS_MAX_INSTANCES; i++) {
```

Comment:
this will cause compassmot to always fail if we don't have 3 compasses (I tested this by changing SITL to have 2 compasses, and found compassmot did indeed fail). We need a check for whether there is a driver for this instance in this loop

Reply:
I'd actually prefer a FORALL_COMPASS_AVAILABLE(i) macro, much like we have FOREACH_I2C_EXTERNAL(i), it would make for easier to read code, and quite a bit smaller patch (as no indent changes)